#! /bin/sh /usr/share/dpatch/dpatch-run
## fix_admin.py.dpatch by  <giskard@localhost.localdomain>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Debian Bug #359721
## DP: Improve the search method in the admin page wich list members



@DPATCH@
diff -urNad mailman-2.1.8~/Mailman/Cgi/admin.py mailman-2.1.8/Mailman/Cgi/admin.py
--- mailman-2.1.8~/Mailman/Cgi/admin.py        2006-08-09 21:19:38.000000000 +0200
+++ mailman-2.1.8/Mailman/Cgi/admin.py 2006-08-09 21:23:01.000000000 +0200
@@ -944,7 +944,7 @@
         for letter in digits + lowercase:
             if not buckets.get(letter):
                 continue
-            url = adminurl + '/members?letter=%s' % letter
+            url = adminurl + '/members?findmember=%s&letter=%s' %(urllib.quote(regexp) ,letter)
             if letter == bucket:
                 show = Bold('[%s]' % letter.upper()).Format()
             else:

